packages/components/src/templates/next/components/internal/ScrollToTop.tsx

```diff
@@ -0,0 +1,19 @@
+"use client"
+
+import { useEffect } from "react"
+
+/*
+  NOTE: This is a temporary fix until we have a better solution.
+
+  Fix with issue w/ "next/Link" not scrolling to the top when clicking on <Link />
+  if we have a header with "sticky" that's not in its sticky position.
+  Ref: https://github.com/vercel/next.js/issues/45187#issuecomment-1639518030
+*/
+export const ScrollToTop = () => {
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (window) window.scrollTo(0, 0)
+  }, [])
```

Comment:
not too sure if this works? won't the empty dependency array check once at initialisation and not check any further? 

as a matter of style, i think we should specify `window` here even though at run-time `window` will be available since it's on the browser

Reply:
yup it will still work and should be empty dependency as this is only to prevent this from running at build-time

According to [nextjs doc](https://nextjs.org/docs/app/building-your-application/deploying/static-exports#browser-apis)
<img width="672" alt="image" src="https://github.com/user-attachments/assets/8c74cc4a-8a93-4959-982d-745ea3dfe82f" />

actually after re-reading this, i realized we can do away with `if (window)` - will make the change